src/namespace.c

```diff
@@ -115,6 +144,56 @@ cc_oci_str_to_ns (const char *str)
 }
 
 /**
+ * Join to a specific namespace
+ *
+ * \param ns \ref oci_cfg_namespace.
+ *
+  * \return \c true on success, else \c false.
+ */
+gboolean cc_oci_ns_join(struct oci_cfg_namespace *ns)
+{
+	int fd = -1;
+	int saved;
+	const gchar *type;
+
+	type = cc_oci_ns_to_str (ns->type);
+
+	fd = open (ns->path, O_RDONLY);
+
+	if (fd < 0) {
+		saved = errno;
+		g_critical ("failed to open %s", ns->path);
+		goto err;
+	}
+
+	/* join an existing namespace */
+	if (setns (fd, ns->type) < 0) {
+		saved = errno;
+		g_critical ("failed to join to %s namespace %s",
+			type ? type : "", ns->path);
+		goto err;
+	}
+
+	close (fd);
+
+	/* reset to ensure any subsequent failure is
+	 * handled correctly.
+	 */
+	fd = -1;
```

Comment:
Do you need to set the fd here, since you are immediately returning after this statement.